src/main/java/org/itc/LibroRepositorio.java

```diff
@@ -0,0 +1,32 @@
+package org.itc;
+
+    import org.springframework.stereotype.Component;
+
+    import java.util.ArrayList;
+import java.util.Collection;
+import java.util.List;
+
+    @Component
+    public class LibroRepositorio implements RecursoRepositorio<Libro> {
+        private final List<Libro> libros = new ArrayList<>();
+
+        @Override
+        public void agregar(Libro libro) {
+            libros.add(libro);
+        }
+
+        @Override
+        public void eliminar(Libro libro) {
+            libros.remove(libro);
+        }
+
+        @Override
+        public Collection<Libro> buscar(String nombre) {
+            return new ArrayList<>(libros);
```

Comment:
Por qué no estan filtrando los datos?

```suggestion
            return new libros.stream().filter(libro -> libro.coincideConCriterio(criterio)).toList();
```